lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -359,9 +359,9 @@ function _campaign_resource_reportback($nid, $values) {
     watchdog('dosomething_api', '_campaign_resource_reportback:' . json_encode($values));
   }
 
-  $transaction_id = $_SERVER['HTTP_X_REQUEST_ID'];
+  $headers = getallheaders();
 
-  if (! $transaction_id) {
+  if (! isset($headers['X-Request-Id'])) {
     $rogue_reportback = dosomething_rogue_send_reportback_to_rogue($values, $user);
```

Comment:
I know this looks weird but the [`getallheaders()`](http://php.net/manual/en/function.apache-request-headers.php) returns the headers that are all caps into only first letter capitalized and the rest lowercase e.g. `X-CSRF-Token` is returned as `X-Csrf-Token` and `X-Request-ID` is returned as `X-Request-Id`